In [32]:
# DATE:
!date \

# SYSTEM INFO:
! system_profiler SPSoftwareDataType

Tue Nov  1 13:39:32 PDT 2016
Software:

    System Software Overview:

      System Version: OS X 10.11.6 (15G1004)
      Kernel Version: Darwin 15.6.0
      Boot Volume: Peach Macbook
      Boot Mode: Normal
      Computer Name: laura spencer’s MacBook Pro
      User Name: laura spencer (shlaura3)
      Secure Virtual Memory: Enabled
      System Integrity Protection: Enabled
      Time since boot: 11 days 14:29



---

# Learning the process of BLAST

In [1]:
!/Applications/ncbi-blast-2.5.0+/bin/blastp

BLAST query/options error: Either a BLAST database or subject sequence(s) must be specified
Please refer to the BLAST+ user manual.


## Viewing geoduck data
Steven & Brent [sampled geoduck in 2015](http://onsnetwork.org/halfshell/2015/08/11/big-day-big-clam/); this is me getting a glimpse of the raw data of said _p. generosa_ genome; BGI did the sequencing, and the files are [here](http://owl.fish.washington.edu/P_generosa_genome_assemblies_BGI/20160512/). 

In [4]:
! curl http://owl.fish.washington.edu/P_generosa_genome_assemblies_BGI/20160512/Panopea_generosa.scafSeq \
| head 

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0>scaffold1 26.6
ACTAACAGACGATCATTAAATACAGTCATACAGAGCAAATGAATTAAGCAATAACCTTGTCTTTAAAGGCCACTTTTGCTCATTCCCTtgtcttgctgct
taaaacaagcttgactgtagatgtatccttaaaggctctcacaaagcattataaccgttatatcaattcattctatgatgctttaGCCCTTTACTCCATA
CATACGTGTTTTCCCGCACCAATCACAGAAtgcctttagaaactgttttccactctgtcagttaatataaacataagttacccaataattcaaaagccat
ctatattagaggtgtccaaaggcgtctctattaggtgtccaaaggcgtcgatatgaagtaaagggttaaatcaaTTTTTTGTAACCCCATCTGTTAATAT
ACATATATCTAGCTGATGTAATTTACTATGACCTTAAAATGATTTTCGTAAACTCAGTAAAGCTCATTTTAACAAGTTTGCAAAATATATATTATGCAAG
GCTGTGTCGGGCATTATATCATTTTTTGCCAATGAGTTGAATAAATTTATCATTGACTTTAAAAAGGTTTCCATCTATCTTAGAAATAGATATGACGGAC
ACTATGGTTTTACCATGACAAGTCAACaaatcgttatcatgtaccattaaatctgatactatattggttttccaggaaattgttttaattccatggcact
cgtaaatattCACTTGTTTTCTGATGTAACTTGTTTTTGTT

## Prepping directory folders

In [1]:
pwd

'/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/Jupyter'

In [2]:
cd ..

/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project


In [3]:
ls

Jupyter/


In [6]:
mkdir data/ analyses/ scripts/

mkdir: data/: File exists


In [7]:
ls

Jupyter/  analyses/ data/     scripts/


In [8]:
cd data/

/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/data


## Downloading geoduck genome file into local directory

In [10]:
!wget http://owl.fish.washington.edu/P_generosa_genome_assemblies_BGI/20160512/Panopea_generosa.scafSeq

/bin/sh: wget: command not found


_**Interesting note: the command `wget` is not installed on Macs**_

In [15]:
!curl -O http://owl.fish.washington.edu/P_generosa_genome_assemblies_BGI/20160512/Panopea_generosa.scafSeq.zip

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  350M  100  350M    0     0   898k      0  0:06:40  0:06:40 --:--:--  909k


**Notes:**  
* Remember to use `!` before all commands (except for `cd` `pwd` `ls` etc.?)  
* -o (lowercase o) the result will be saved in the filename provided in the command line  
* -O (uppercase O) the filename in the URL will be taken and it will be used as the filename to store the result
* Helpful `curl` tips: http://www.thegeekstuff.com/2012/04/curl-examples/

In [32]:
ls

Panopea_generosa.scafSeq      Panopea_generosa.scafSeq.zip


---

# Experimenting with commands to learn the ropes

In [48]:
!head Panopea_generosa.scafSeq

>scaffold1 26.6
ACTAACAGACGATCATTAAATACAGTCATACAGAGCAAATGAATTAAGCAATAACCTTGTCTTTAAAGGCCACTTTTGCTCATTCCCTtgtcttgctgct
taaaacaagcttgactgtagatgtatccttaaaggctctcacaaagcattataaccgttatatcaattcattctatgatgctttaGCCCTTTACTCCATA
CATACGTGTTTTCCCGCACCAATCACAGAAtgcctttagaaactgttttccactctgtcagttaatataaacataagttacccaataattcaaaagccat
ctatattagaggtgtccaaaggcgtctctattaggtgtccaaaggcgtcgatatgaagtaaagggttaaatcaaTTTTTTGTAACCCCATCTGTTAATAT
ACATATATCTAGCTGATGTAATTTACTATGACCTTAAAATGATTTTCGTAAACTCAGTAAAGCTCATTTTAACAAGTTTGCAAAATATATATTATGCAAG
GCTGTGTCGGGCATTATATCATTTTTTGCCAATGAGTTGAATAAATTTATCATTGACTTTAAAAAGGTTTCCATCTATCTTAGAAATAGATATGACGGAC
ACTATGGTTTTACCATGACAAGTCAACaaatcgttatcatgtaccattaaatctgatactatattggttttccaggaaattgttttaattccatggcact
cgtaaatattCACTTGTTTTCTGATGTAACTTGTTTTTGTTGTTTTTTCAGCCTGAATATAGATAACTATAACNGTTATCATGTACCATTAAATCTGATA
CTATATTGGTTTTCCAGGAAATTGTTTTAATTCCATGGCACTCGTAAatattcacttgttttctgatgtaacttgtttttgttgttttttcagcctgaat


In [49]:
!tail Panopea_generosa.scafSeq

GAGGACAAAATTGAGATTGTTGTGGACGGGAAACAGGAAACATATGAAGACGAGTTCGCGGTCATTAAGGTTAAGGTTTCAGACTTGGACGAAATGAAAA
TACCTTATGAGTTAGTTTAACCGCGGTACGCCAATGTATGATATACTTATACCAGTAAATGATTAGGAGGAATATTAATGAGCGAGAAAGTAGAATTGAT
TGTTGTTGATATGGTTGACGAGGAAATTAGTACTAAGTTCCAATTGTGTCCTGAAGACTACTCGAAAGTGTATACACTGCAGTTGTTCAAACAGGACTAT
AAAAAGGACACCAAAGAATGGGTCAGTGATGCGGAAACCACCAAACAATACGATGCGGATATGAAAGAGTTGGGTAACCCTAGCGTGGGTGATAAGGTCA
ATGTGTTCGTTTCAAACAACGGCAAAGCGTATTTGACTGAAAAAACATTCGTGGAAACGGTGAAACCTAGCGGGAAAGATGCGAACAAATTGTTCACAAA
TTGCGAAATTGAATCGTTCCATGATTCGCCAAAAGGGCGACGCGTGATTGTCAAGTATAAAAAGAATTTTTACGAGTTTAGTTTCAATTATTCCGTTTGG
ATTCCGTCAATTGAGAAGTACGTTCCAAATGATGCCAAAATGGAAAAGGCAAAAGTGCGATTCAACGATATTTTTGAACATTCGTTGATTACGTGGGAAA
ACGCACTAGAAACGGACGCGGATGGTGTTAGCATCCATAAGGGTGTTAAAATGAACTGCTTGGTTCGCAAAAACGAATTGGACGCAACTGGTCAAACAGG
TTGGTTGGAACCGTTGAAAATTGACGAACCAGATGTTGACGAAGACGACCTTCCGTTCTAAAAATAGTTACAAAAAAGTGTAAGTTTAGGCTTGCACTTT
TTATTTTACCGTGGTAAGATATACTCAACAAGTTGAGTGA


**Unsure how to navigate in the `less` program**

In [50]:
!less Panopea_generosa.scafSeq

>scaffold1 26.6
ACTAACAGACGATCATTAAATACAGTCATACAGAGCAAATGAATTAAGCAATAACCTTGTCTTTAAAGGCCACTTTTGCT CATTCCCTtgtcttgctgct
taaaacaagcttgactgtagatgtatccttaaaggctctcacaaagcattataaccgttatatcaattcattctatgatg ctttaGCCCTTTACTCCATA
CATACGTGTTTTCCCGCACCAATCACAGAAtgcctttagaaactgttttccactctgtcagttaatataaacataagtta cccaataattcaaaagccat
ctatattagaggtgtccaaaggcgtctctattaggtgtccaaaggcgtcgatatgaagtaaagggttaaatcaaTTTTTT GTAACCCCATCTGTTAATAT
ACATATATCTAGCTGATGTAATTTACTATGACCTTAAAATGATTTTCGTAAACTCAGTAAAGCTCATTTTAACAAGTTTG CAAAATATATATTATGCAAG
GCTGTGTCGGGCATTATATCATTTTTTGCCAATGAGTTGAATAAATTTATCATTGACTTTAAAAAGGTTTCCATCTATCT TAGAAATAGATATGACGGAC
ACTATGGTTTTACCATGACAAGTCAACaaatcgttatcatgtaccattaaatctgatactatattggttttccaggaaat tgttttaattccatggcact
cgtaaatattCACTTGTTTTCTGATGTAACTTGTTTTTGTTGTTTTTTCAGCCTGAATATAGATAACTATAACNGTTATC ATGTACCATTAAATCTGATA
CTATATTGGTTTTCCAGGAAATTGTTTTAATTCCATGGCACTCGTAAatattcacttgttttctgatgtaacttgttttt gttgttttttcagcctgaat
atagataactataacATGTTACTAGGAATGGGGTTCCCTAAGGGAGCTTCAGAAGAG

**Removing first row (metadata), viewing only first 5 rows of sequences**

In [59]:
!tail -n +2 Panopea_generosa.scafSeq | head -n 5

ACTAACAGACGATCATTAAATACAGTCATACAGAGCAAATGAATTAAGCAATAACCTTGTCTTTAAAGGCCACTTTTGCTCATTCCCTtgtcttgctgct
taaaacaagcttgactgtagatgtatccttaaaggctctcacaaagcattataaccgttatatcaattcattctatgatgctttaGCCCTTTACTCCATA
CATACGTGTTTTCCCGCACCAATCACAGAAtgcctttagaaactgttttccactctgtcagttaatataaacataagttacccaataattcaaaagccat
ctatattagaggtgtccaaaggcgtctctattaggtgtccaaaggcgtcgatatgaagtaaagggttaaatcaaTTTTTTGTAACCCCATCTGTTAATAT
ACATATATCTAGCTGATGTAATTTACTATGACCTTAAAATGATTTTCGTAAACTCAGTAAAGCTCATTTTAACAAGTTTGCAAAATATATATTATGCAAG
tail: stdout: Broken pipe


**Experimenting with `grep` and color coding**

In [66]:
!tail -n +2 Panopea_generosa.scafSeq | head -n 5 \
| grep -C2 "AAATACA" --color=auto

ACTAACAGACGATCATTAAATACAGTCATACAGAGCAAATGAATTAAGCAATAACCTTGTCTTTAAAGGCCACTTTTGCTCATTCCCTtgtcttgctgct
taaaacaagcttgactgtagatgtatccttaaaggctctcacaaagcattataaccgttatatcaattcattctatgatgctttaGCCCTTTACTCCATA
CATACGTGTTTTCCCGCACCAATCACAGAAtgcctttagaaactgttttccactctgtcagttaatataaacataagttacccaataattcaaaagccat
tail: stdout: Broken pipe


In [67]:
!file Panopea_generosa.scafSeq

Panopea_generosa.scafSeq: ASCII text


---

# Exploring data to prepare for Blast protocol

Trying to really manipulate the genome data in order to Blast in comparison to transcriptome data. First let's see the first 20 rows of the data:**

In [1]:
pwd

'/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/Jupyter'

In [4]:
cd ..

/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project


In [5]:
cd data

/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/data


In [6]:
ls

Panopea_generosa.scafSeq      Panopea_generosa.scafSeq.zip


In [26]:
ls

Geo-70k-scaff-annotations-merged-table.tabular
Geo-v3-join-uniprot-all0916-condensed.tab
Geoduck-transcriptome-v3.fa.zip
Geoduck-transcriptome-v3.fa_db.nhr
Geoduck-transcriptome-v3.fa_db.nin
Geoduck-transcriptome-v3.fa_db.nsq
Geoduck-transcriptome-v3annotated.fa*
Panopea_generosa_No-Line-Breaks.scafSeq.fai
Panopea_generosa_ScafSeq.genome
Panopea_generosa_scaff-70k.scafSeq
Panopea_generosa_scaff-70k.scafSeq.fai
Panopea_generosa_scaff-70k_db.nhr
Panopea_generosa_scaff-70k_db.nin
Panopea_generosa_scaff-70k_db.nsq
Phel_countdata.txt
Phel_transcriptome.fasta
hairpin.fa


Checking out the # of scaffolds

In [9]:
! grep -c "^>" Panopea_generosa.scafSeq

1296135


---

# Subsetting only >70k bp scaffolds in genome

**Ok, there are nearly 1.3 million scaffolds. Steven suggested paring this data down to only very large scaffolds, such as on those >70k. This script was borrowed from [Steven's notebbok](https://sr320.github.io/student-fish546-2016/data-description/)**

In [10]:
!awk '!/^>/ { next } { getline seq } length(seq) >= 70000 { print $0 "\n" seq }' \
../data/Panopea_generosa.scafSeq > ../data/Panopea_generosa_scaff-70k.scafSeq

In [11]:
ls

Panopea_generosa.scafSeq            Panopea_generosa_scaff-70k.scafSeq
Panopea_generosa.scafSeq.zip


In [16]:
! wc Panopea_generosa_scaff-70k.scafSeq

       0       0       0 Panopea_generosa_scaff-70k.scafSeq


**Hmm... the new file doesn't contain anything; turns out this is because there are "hard line-breaks" within scaffolds, so the search isn't finding any super long contigs. Now using a script (from Steven) to remove these line breaks:**

In [20]:
!awk '!/^>/ { printf "%s", $0; n = "\n" } /^>/ { print n $0; n = "" } END { printf "%s", n }' \
../data/Panopea_generosa.scafSeq > ../data/Panopea_generosa_No-Line-Breaks.scafSeq

In [14]:
!head -n 5 ../data/Panopea_generosa_No-Line-Breaks.scafSeq

>scaffold1 26.6
ACTAACAGACGATCATTAAATACAGTCATACAGAGCAAATGAATTAAGCAATAACCTTGTCTTTAAAGGCCACTTTTGCTCATTCCCTtgtcttgctgcttaaaacaagcttgactgtagatgtatccttaaaggctctcacaaagcattataaccgttatatcaattcattctatgatgctttaGCCCTTTACTCCATACATACGTGTTTTCCCGCACCAATCACAGAAtgcctttagaaactgttttccactctgtcagttaatataaacataagttacccaataattcaaaagccatctatattagaggtgtccaaaggcgtctctattaggtgtccaaaggcgtcgatatgaagtaaagggttaaatcaaTTTTTTGTAACCCCATCTGTTAATATACATATATCTAGCTGATGTAATTTACTATGACCTTAAAATGATTTTCGTAAACTCAGTAAAGCTCATTTTAACAAGTTTGCAAAATATATATTATGCAAGGCTGTGTCGGGCATTATATCATTTTTTGCCAATGAGTTGAATAAATTTATCATTGACTTTAAAAAGGTTTCCATCTATCTTAGAAATAGATATGACGGACACTATGGTTTTACCATGACAAGTCAACaaatcgttatcatgtaccattaaatctgatactatattggttttccaggaaattgttttaattccatggcactcgtaaatattCACTTGTTTTCTGATGTAACTTGTTTTTGTTGTTTTTTCAGCCTGAATATAGATAACTATAACNGTTATCATGTACCATTAAATCTGATACTATATTGGTTTTCCAGGAAATTGTTTTAATTCCATGGCACTCGTAAatattcacttgttttctgatgtaacttgtttttgttgttttttcagcctgaatatagataactataacATGTTACTAGGAATGGGGTTCCCTAAGGGAGCTTCAGAAGAGGCACTCAGACAGGGAAACAATGATGTC

**Looks like ALL line breaks have gone away now, b/c when I ask for the first 5 lines it produces many more. I'll try the >70k script again, see what happens**

In [23]:
!awk '!/^>/ { next } { getline seq } length(seq) >= 70000 { print $0 "\n" seq }' \
../data/Panopea_generosa_No-Line-Breaks.scafSeq > ../data/Panopea_generosa_scaff-70k.scafSeq

In [25]:
! grep -c "^>" ../data/Panopea_generosa_scaff-70k.scafSeq

20


**Great! I've subselected the largest 20 scaffolds (from ~1.3M)**  
The new data file with only >70k scaffolds is called Panopea_generosa_scaff-70k.scafSeq

---

# For real this time - Blasting transcriptome against genome 

**Now I'm going to blast the _p. generosa_ transcriptome Steven has been working on against the _p. generosa_ genome. I'm using the blastn program because it compares nucleotide-nucleotide.**

In [7]:
!/Applications/ncbi-blast-2.5.0+/bin/blastn

BLAST query/options error: Either a BLAST database or subject sequence(s) must be specified
Please refer to the BLAST+ user manual.


**Downloading _p. generosa_ transcriptome**

In [8]:
cd ../data

/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/data


In [9]:
!curl -O https://github.com/sr320/paper-pano-go/raw/master/data-results/Geoduck-transcriptome-v3.fa.zip

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   171  100   171    0     0    313      0 --:--:-- --:--:-- --:--:--   316


**unzipped the file by using the gui directory**

In [15]:
ls

Geoduck-transcriptome-v3.fa.zip
Geoduck-transcriptome-v3.fa.zip.cpgz
Panopea_generosa.scafSeq
Panopea_generosa.scafSeq.zip
Panopea_generosa_No-Line-Breaks.scafSeq
Panopea_generosa_scaff-70k.scafSeq


**Building a databse out of the _p. generosa_ genome file, since I'll be blasting the transcriptome against the genome**

In [16]:
!/Applications/ncbi-blast-2.5.0+/bin/makeblastdb \
-in ../data/Panopea_generosa_scaff-70k.scafSeq \
-dbtype nucl \
-out ../data/Panopea_generosa_scaff-70k_db



Building a new DB, current time: 10/20/2016 11:29:24
New DB name:   /Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/data/Panopea_generosa_scaff-70k_db
New DB title:  ../data/Panopea_generosa_scaff-70k.scafSeq
Sequence type: Nucleotide
Keep MBits: T
Maximum file size: 1000000000B
Adding sequences from FASTA; added 20 sequences in 0.0768011 seconds.


In [17]:
ls

Geoduck-transcriptome-v3.fa.zip
Geoduck-transcriptome-v3.fa.zip.cpgz
Panopea_generosa.scafSeq
Panopea_generosa.scafSeq.zip
Panopea_generosa_No-Line-Breaks.scafSeq
Panopea_generosa_scaff-70k.scafSeq
Panopea_generosa_scaff-70k_db.nhr
Panopea_generosa_scaff-70k_db.nin
Panopea_generosa_scaff-70k_db.nsq


In [20]:
!/Applications/ncbi-blast-2.5.0+/bin/blastn \
-query ../data/Geoduck-transcriptome-v3.fa.zip.cpgz \
-db ../data/Panopea_generosa_scaff-70k_db \
-outfmt 6  \
-out ../analyses/pgenerosa-transcrv3-blastn-scaff70k-01.tab


Error: NCBI C++ Exception:
    T0 "/Users/coremake/release_build/build/PrepareRelease_IntelMAC-Clang36_JSID_01_40204_130.14.22.10_9008__PrepareRelease_IntelMAC-Clang36_1473441060/c++/compilers/unix/../../src/objtools/readers/fasta.cpp", line 2178: Error: ncbi::objects::CFastaReader::PostWarning() - CFastaReader: Near line 1, there's a line that doesn't look like plausible data, but it's not marked as defline or comment. (m_Pos = 1)



In [24]:
!head -n 100 Geoduck-transcriptome-v3.fa.zip.cpgz

��P�J�@���{�̴3݂�"��ŋǴͶ�mgȤ�~��
ck%T��՗������m��wVo�������!I��������

**Something is not right with the transcriptome file. Turns out while unzipping it created the .cpgz zip file (interesting read on .cpgz)[http://osxdaily.com/2013/02/13/open-zip-cpgz-file/]. I'm going to download the .zip file again, then try to unzip at command line**

In [25]:
ls

Geoduck-transcriptome-v3.fa.zip
Geoduck-transcriptome-v3.fa.zip.cpgz
Panopea_generosa.scafSeq
Panopea_generosa.scafSeq.zip
Panopea_generosa_No-Line-Breaks.scafSeq
Panopea_generosa_scaff-70k.scafSeq
Panopea_generosa_scaff-70k_db.nhr
Panopea_generosa_scaff-70k_db.nin
Panopea_generosa_scaff-70k_db.nsq


In [28]:
!rm Geoduck-transcriptome-v3.fa.zip

rm: Geoduck-transcriptome-v3.fa.zip: No such file or directory


In [30]:
ls

Panopea_generosa.scafSeq
Panopea_generosa.scafSeq.zip
Panopea_generosa_No-Line-Breaks.scafSeq
Panopea_generosa_scaff-70k.scafSeq
Panopea_generosa_scaff-70k_db.nhr
Panopea_generosa_scaff-70k_db.nin
Panopea_generosa_scaff-70k_db.nsq


In [31]:
!curl -O https://github.com/sr320/paper-pano-go/raw/master/data-results/Geoduck-transcriptome-v3.fa.zip

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   171  100   171    0     0    297      0 --:--:-- --:--:-- --:--:--   304


In [4]:
cd ../data

/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/data


In [5]:
ls

Geoduck-transcriptome-v3.fa.zip
Panopea_generosa.scafSeq
Panopea_generosa.scafSeq.zip
Panopea_generosa_No-Line-Breaks.scafSeq
Panopea_generosa_scaff-70k.scafSeq
Panopea_generosa_scaff-70k_db.nhr
Panopea_generosa_scaff-70k_db.nin
Panopea_generosa_scaff-70k_db.nsq


In [6]:
!unzip Geoduck-transcriptome-v3.fa.zip

Archive:  Geoduck-transcriptome-v3.fa.zip
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of Geoduck-transcriptome-v3.fa.zip or
        Geoduck-transcriptome-v3.fa.zip.zip, and cannot find Geoduck-transcriptome-v3.fa.zip.ZIP, period.


**It's clear that the file downloaded via `curl` is corrupted. I deleted it, and will download again via GUI, then `unzip` and `blast` as previously attempted**

In [3]:
cd ../data

/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/data


In [5]:
ls

Geoduck-transcriptome-v3.fa.zip
Panopea_generosa.scafSeq
Panopea_generosa.scafSeq.zip
Panopea_generosa_No-Line-Breaks.scafSeq
Panopea_generosa_scaff-70k.scafSeq
Panopea_generosa_scaff-70k_db.nhr
Panopea_generosa_scaff-70k_db.nin
Panopea_generosa_scaff-70k_db.nsq


In [6]:
!unzip Geoduck-transcriptome-v3.fa.zip

Archive:  Geoduck-transcriptome-v3.fa.zip
  inflating: Geoduck-transcriptome-v3.fa  


In [7]:
!/Applications/ncbi-blast-2.5.0+/bin/blastn

BLAST query/options error: Either a BLAST database or subject sequence(s) must be specified
Please refer to the BLAST+ user manual.


In [8]:
!/Applications/ncbi-blast-2.5.0+/bin/blastn \
-query ../data/Geoduck-transcriptome-v3.fa \
-db ../data/Panopea_generosa_scaff-70k_db \
-outfmt 6  \
-out ../analyses/pgenerosa-transcrv3-blastn-scaff70k-01.tab

In [14]:
cd ../analyses

/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/analyses


In [15]:
ls

pgenerosa-transcrv3-blastn-scaff70k-01.tab


In [16]:
!head pgenerosa-transcrv3-blastn-scaff70k-01.tab

comp100323_c0_seq1	scaffold10970	95.161	62	3	0	3	64	58201	58262	5.51e-22	99.0
comp100333_c1_seq1	scaffold10970	88.800	125	10	3	1	121	99916	100040	2.05e-37	150
comp100468_c0_seq1	scaffold9504	100.000	28	0	0	206	233	59077	59104	5.04e-08	52.8
comp101573_c0_seq1	scaffold32578	98.341	422	6	1	1	422	55507	55927	0.0	739
comp101587_c0_seq1	scaffold30278	100.000	29	0	0	275	303	60566	60538	1.86e-08	54.7
comp102059_c0_seq1	scaffold71773	81.875	160	23	5	1	158	55978	55823	2.38e-31	130
comp102699_c0_seq1	scaffold15463	86.765	68	9	0	91	158	6270	6203	2.79e-15	76.8
comp103000_c0_seq1	scaffold59644	98.305	118	2	0	105	222	2781	2664	9.63e-55	207
comp103462_c0_seq1	scaffold26337	83.784	111	6	2	156	254	134703	134813	1.03e-20	95.3
comp103462_c0_seq2	scaffold26337	83.784	111	6	2	185	283	134703	134813	1.14e-20	95.3


In [18]:
! grep -c "^comp" pgenerosa-transcrv3-blastn-scaff70k-01.tab

3366


**If blasted correctly, there are 3,366 matches**

---

# Re-blasting per Steven's feedback
Steven provided [video feedback](https://uw.hosted.panopto.com/Panopto/Pages/Viewer.aspx?id=f3692c33-b29b-44d9-ac2b-ac1540b518f1) of my work thus far. He suggested blasting again using modified commands, namely restricting hits with e-values < 1e-20, and allowing for mutliple hits on a contig. Here's me re-doing the blast

In [1]:
!/Applications/ncbi-blast-2.5.0+/bin/blastn

BLAST query/options error: Either a BLAST database or subject sequence(s) must be specified
Please refer to the BLAST+ user manual.


In [6]:
cd ../data

[Errno 2] No such file or directory: '../data | ls'
/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/data


In [7]:
ls

Geoduck-transcriptome-v3.fa
Geoduck-transcriptome-v3.fa.zip
Geoduck-transcriptome-v3.fa_db.nhr
Geoduck-transcriptome-v3.fa_db.nin
Geoduck-transcriptome-v3.fa_db.nsq
Panopea_generosa.scafSeq
Panopea_generosa.scafSeq.zip
Panopea_generosa_No-Line-Breaks.scafSeq
Panopea_generosa_scaff-70k.scafSeq
Panopea_generosa_scaff-70k_db.nhr
Panopea_generosa_scaff-70k_db.nin
Panopea_generosa_scaff-70k_db.nsq
Phel_countdata.txt
Phel_transcriptome.fasta


In [8]:
!/Applications/ncbi-blast-2.5.0+/bin/blastn \
-query ../data/Geoduck-transcriptome-v3.fa \
-db ../data/Panopea_generosa_scaff-70k_db \
-task blastn \
-evalue 1e-20 \
-outfmt 6 \
-max_target_seqs 1 \
-num_threads 6 \
-out ../analyses/pgenerosa-transcrv3-blastn-scaff70k-01.tab

Woops, I accidentally overwrote the previous blast ouput file. Oh well! Good lesson to learn eary in the game.

In [9]:
!head ../analyses/pgenerosa-transcrv3-blastn-scaff70k-01.tab

comp100007_c0_seq1	scaffold26337	70.435	230	44	7	351	556	55193	55422	2.67e-21	98.7
comp100064_c1_seq1	scaffold71773	82.192	146	23	3	76	220	36938	36795	2.95e-33	136
comp100102_c0_seq1	scaffold4463	79.444	180	28	5	1	175	12886	13061	1.82e-35	143
comp100151_c0_seq1	scaffold34940	79.866	149	29	1	1	148	28989	28841	1.22e-31	131
comp100168_c0_seq1	scaffold26337	68.222	343	83	11	412	744	116776	117102	2.22e-22	102
comp100173_c0_seq1	scaffold19489	76.923	130	20	2	2	124	30526	30652	2.02e-22	100
comp100177_c1_seq1	scaffold5354	70.279	323	58	6	5	289	23010	23332	1.72e-39	158
comp100177_c1_seq1	scaffold5354	74.317	183	28	3	5	168	31421	31603	1.24e-28	122
comp100181_c0_seq1	scaffold11875	80.583	103	11	1	232	334	31235	31142	3.35e-22	100
comp100181_c0_seq2	scaffold11875	80.583	103	11	1	102	204	31235	31142	1.96e-22	100


In [11]:
! grep -c "^comp" ../analyses/pgenerosa-transcrv3-blastn-scaff70k-01.tab

22781


**This time, by allowing for multiple matches in the blast I got 22,781 contigs > 70k**

In [18]:
# Using a different method of counting the scaffold hits - same result
!wc -l ../analyses/pgenerosa-transcrv3-blastn-scaff70k-01.tab

   22781 ../analyses/pgenerosa-transcrv3-blastn-scaff70k-01.tab


In [19]:
cd ../scripts

/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/scripts


Steven used a perl script file to transform the .tab blast output to gff for visualization, so I will do the same

In [22]:
# downloading the perl script from Steven's GitHub
!curl -O https://raw.githubusercontent.com/sr320/student-fish546-2016/master/scripts/2_Blast2Gff.pl 

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  7427  100  7427    0     0   4376      0  0:00:01  0:00:01 --:--:--  233k


In [21]:
ls

2_Blast2Gff.pl      Helpful-Scripts.md  count_fasta.pl


In [23]:
!perl ../scripts/2_Blast2Gff.pl \
-i ../analyses/pgenerosa-transcrv3-blastn-scaff70k-01.tab \
-s "something" \
-o ../analyses/pgenerosa-transcrv3-blastn-scaff70k-01.gff \
-p "gene-blast" \
-d "geoduck-scaff-70k"

In [24]:
!head ../analyses/pgenerosa-transcrv3-blastn-scaff70k-01.gff

scaffold26337	blastn:geoduck-scaff-70k	blastn	55193	55422	2.67e-21	+	.	comp100007_c0_seq1	
scaffold71773	blastn:geoduck-scaff-70k	blastn	36938	36795	2.95e-33	-	.	comp100064_c1_seq1	
scaffold4463	blastn:geoduck-scaff-70k	blastn	12886	13061	1.82e-35	+	.	comp100102_c0_seq1	
scaffold34940	blastn:geoduck-scaff-70k	blastn	28989	28841	1.22e-31	-	.	comp100151_c0_seq1	
scaffold26337	blastn:geoduck-scaff-70k	blastn	116776	117102	2.22e-22	+	.	comp100168_c0_seq1	
scaffold19489	blastn:geoduck-scaff-70k	blastn	30526	30652	2.02e-22	+	.	comp100173_c0_seq1	
scaffold5354	blastn:geoduck-scaff-70k	blastn	23010	23332	1.72e-39	+	.	comp100177_c1_seq1	
scaffold5354	blastn:geoduck-scaff-70k	blastn	31421	31603	1.24e-28	+	.	comp100177_c1_seq1	
scaffold11875	blastn:geoduck-scaff-70k	blastn	31235	31142	3.35e-22	-	.	comp100181_c0_seq1	
scaffold11875	blastn:geoduck-scaff-70k	blastn	31235	31142	1.96e-22	-	.	comp100181_c0_seq2	


In [27]:
# confirming same # of lines, should still be 22781
!wc -l ../analyses/pgenerosa-transcrv3-blastn-scaff70k-01.gff

   22781 ../analyses/pgenerosa-transcrv3-blastn-scaff70k-01.gff


In [33]:
# In comparison, the total genome file has the following number of scaffolds 
! grep -c "^>" ../data/Panopea_generosa.scafSeq

1296135


**Loaded file into the IGV [software tool](http://software.broadinstitute.org/software/igv/feature_track_options) to get a visual of the scaffolds and where transcriptome matches are located**

All 20 scaffolds:

![All scaffolds](../images/2016-11-3_Feature-track-all-scaffolds.png)

Scaffold # 71773

![Scaffold 71773](../images/2016-11-3_feature-track-scaff71773.png)

-----

# Merging blastn results with geoduck transcriptome using Galaxy online tool to connect nucleotides sequences to scaffold/contig IDs

Uploading both files

![upload both files](../images/2016-11-3_upload-files.png)

Converting the transcriptome file from fasta to tab format

![Convert transciptome file from fasta to tab](../images/2016-11-3_transc-fasta-to-tab.png)

Removing excess data from transcriptome file 

![Removing excess data from transcriptome file](../images/2016-11-3_transc-remove-excess.png)

Merging files

![Merging files](../images/2016-11-3_merge-files.png)

In [11]:
# Downloaded file; let's take a look at it:
!head ../analyses/2016-11-3_Geo70k-scaff-transcrip-sequences.tabular

comp100007_c0_seq1	TTGAAGAGTCTAATAATGGATCGGTAAATGAAGCTTTATGTTCTTAAAAGAGGTATATCTTAGACTTGAAGATGATTATTTAGAAATCTGTAAAAGGTCGAGGAAACTGGACACTGTTTAGGGAATTGCTACCTGAATCACAAGACTTTTGAAACGAAAAGTGCATACCTCATAGTTTTCACACTACACTTATGTTAGCAGCAATGGTAGACATGTGTCTTTGATGGATCCTATTGTGCTACCATTCTTTCTATAGGACTCAAATTAGTGAACCAATGATTCATTTTATGTATTATTATTATATGCATATGCAAATCTCCGACTATGCAATGTGTATTCATGTAGCGCTGTATTTGAGCACATTTGGCAGATCTTCTTTCAATTATATATGACTGTTTGTGATCAATACTACGATAAAAAAATCTGAAAATAGCAAAATTTGTTAGCAGTATTATCGGAAGCACAACAAATTATTCTATATGACTGTGAACTTTCTTAATATGTTTACAGTTCATAATTTATGGGGCGAATAACATTACTCAATGAACGTTCCTCGCCATGCATAGAAAATACATCATAAATATATGTCAACTCTGGAAAGTCACCAGTTCCCATGTATCGTGGTTACTGTGATATTTTCTTATGTTTACAGTTCAGCTTGATAAAAATATAGCAGATCTGAAAAAAAAAATGATCTGTATTCAAAGATTTTCGTTTTAAAGGCGATAGTTATA	scaffold26337	70.435	230	44	7	351	556	55193	55422	2.67e-21	98.7
comp100064_c1_seq1	TACATGGGAATAATCATTTACATGCGTGCTTTAACAAAATATCATCTGAATGTATGGCATTGAAACGTAACACAATGAGGCCCACCCAAGGGTTTTGGAGATCTGGAAAACATGGCCAATTATTTTAGATATCTAGGGAACATTGACCATTACTTTCAGGGAT

---

# Merging my scaff70k transcriptome sequences with annotations

First, downloaded [p. generosa annotated transcriptome data file](https://github.com/sr320/paper-pano-go/blob/master/data-results/Geoduck-transcriptome-v3annotated.fa.zip) from Steven's [paper.pano.go repo](https://github.com/sr320/paper-pano-go). Let's check it out:

In [4]:
ls

Geoduck-transcriptome-v3.fa.zip
Geoduck-transcriptome-v3.fa_db.nhr
Geoduck-transcriptome-v3.fa_db.nin
Geoduck-transcriptome-v3.fa_db.nsq
Geoduck-transcriptome-v3annotated.fa*
Panopea_generosa_No-Line-Breaks.scafSeq.fai
Panopea_generosa_ScafSeq.genome
Panopea_generosa_scaff-70k.scafSeq
Panopea_generosa_scaff-70k.scafSeq.fai
Panopea_generosa_scaff-70k_db.nhr
Panopea_generosa_scaff-70k_db.nin
Panopea_generosa_scaff-70k_db.nsq
Phel_countdata.txt
Phel_transcriptome.fasta


In [12]:
!head ../data/Geoduck-transcriptome-v3annotated.fa


>comp100000_c0_seq1
TGAATGTATGTTTGTGAACGTATGTATATGAATGTATGTATGTGAATGCATACCATCTGT
ATAAGTATAATCCGACCGGGAGATGTTTATTCACACAGTTTGGCATTATGACGTTTAACC
TCTGAATTGGCGTCTCTTGCTACTGACCGCTTCACAGTGATGACCCATGTTGTCACTTCT
AATCTATTTATGTATTGCTCTTTTATATTATACTATTAACGCTGTTAAAATACACTACCG
CTAAACAAATAAGAGTTGTGGGTTTGAATCATTGGTGAGTGCAGGAACCTCAGCAGGTCA
TTAAGATTTACGTGTACGTCTATCCTAAACCTACATGTTTCAACTTTGTTGTTTTTCGGT
TTCGTTCTCTGTACACAGCTCTTGAAACATACATAAAATACCATTTTATTAAAAAATATG
TCTCTATTTAATGTTAAAACCTTTTTAAGAAAA
>comp100001_c1_seq1


In [17]:
# checking out the annotated transcriptome contigs; all annotations begin with "PREDICTED" string
!grep -E "PREDICTED" ../data/Geoduck-transcriptome-v3annotated.fa | head -n 20

>comp100065_c0_seq1 PREDICTED Lipoma HMGIC fusion partner homolog [Panopea generosa]
>comp100097_c0_seq1 PREDICTED Importin-11 (Imp11) (Ran-binding protein 11) (RanBP11) [Panopea generosa]
>comp100104_c2_seq1 PREDICTED Sacsin (DnaJ homolog subfamily C member 29) (DNAJC29) [Panopea generosa]
>comp100105_c1_seq1 PREDICTED Protein c-ets-1-B (C-ets-1B) (XE1-B) (Fragment) [Panopea generosa]
>comp100108_c1_seq1 PREDICTED Apolipophorins [Cleaved into: Apolipophorin-2 (Apolipophorin II) (apoLp-2); Apolipophorin-1 (Apolipophorin I) (apoLp-1)] [Panopea generosa]
>comp100109_c0_seq1 PREDICTED Alpha-tectorin [Panopea generosa]
>comp100113_c0_seq1 PREDICTED Beta-1,3-galactosyltransferase bre-5 (EC 2.4.1.-) (Bacillus thuringiensis toxin-resistant protein 5) (Bt toxin-resistant protein 5) [Panopea generosa]
>comp100113_c0_seq2 PREDICTED Beta-1,3-galactosyltransferase 5 (Beta-1,3-GalTase 5) (Beta3Gal-T5) (Beta3GalT5) (b3Gal-T5) (EC 2.4.1.-) (Beta-3-Gx-T5) (UDP-Gal:beta-GlcNAc beta-1,3-galactosyltransf

In [15]:
# How many transcriptome contigs have been annotated
!grep -c -E "PREDICTED" ../data/Geoduck-transcriptome-v3annotated.fa

22961


In [16]:
# compared to total number of transcriptome contigs 
! grep -c "^>" ../data/Geoduck-transcriptome-v3annotated.fa

153982


In [2]:
cd ../data

/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/data


In [5]:
# Turns out the ..v3annotated.fa file might not be the correct one to work with; doesn't have all the associated protein data. 
! curl -O https://raw.githubusercontent.com/sr320/paper-pano-go/master/data-results/Geo-v3-join-uniprot-all0916-condensed.tab 

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 8797k  100 8797k    0     0  3441k      0  0:00:02  0:00:02 --:--:-- 4108k


In [6]:
ls

Geo-v3-join-uniprot-all0916-condensed.tab
Geoduck-transcriptome-v3.fa.zip
Geoduck-transcriptome-v3.fa_db.nhr
Geoduck-transcriptome-v3.fa_db.nin
Geoduck-transcriptome-v3.fa_db.nsq
Geoduck-transcriptome-v3annotated.fa*
Panopea_generosa_No-Line-Breaks.scafSeq.fai
Panopea_generosa_ScafSeq.genome
Panopea_generosa_scaff-70k.scafSeq
Panopea_generosa_scaff-70k.scafSeq.fai
Panopea_generosa_scaff-70k_db.nhr
Panopea_generosa_scaff-70k_db.nin
Panopea_generosa_scaff-70k_db.nsq
Phel_countdata.txt
Phel_transcriptome.fasta


In [8]:
# This is the geoduck transcriptome annotated with detailed info. from the Uniprot database
! head -10 Geo-v3-join-uniprot-all0916-condensed.tab

GeoID	Evalue	SPID	ProteinName	Organism	GO	PFAM
comp100065_c0_seq1	2.00E-08	Q5PRC1	Lipoma HMGIC fusion partner homolog	Danio rerio (Zebrafish) (Brachydanio rerio)		PF10242;
comp100097_c0_seq1	2.00E-37	Q9UI26	Importin-11 (Imp11) (Ran-binding protein 11) (RanBP11)	Homo sapiens (Human)	ribosomal protein import into nucleus [GO:0006610]; transcription factor import into nucleus [GO:0042991]	PF03810;
comp100104_c2_seq1	3.00E-29	Q9NZJ4	Sacsin (DnaJ homolog subfamily C member 29) (DNAJC29)	Homo sapiens (Human)	negative regulation of inclusion body assembly [GO:0090084]; protein folding [GO:0006457]	PF05168;PF00240;
comp100105_c1_seq1	2.00E-73	P18756	Protein c-ets-1-B (C-ets-1B) (XE1-B) (Fragment)	Xenopus laevis (African clawed frog)	positive regulation of endothelial cell migration [GO:0010595]; regulation of angiogenesis [GO:0045765]; transcription, DNA-templated [GO:0006351]	PF00178;
comp100108_c1_seq1	3.00E-59	Q9U943	Apolipophorins [Cleaved into: Apolipophorin-2 (Apolipophorin II) (apoLp-2)

Now let's merge this file with my genome/transcriptome file using Galaxy: 

![Galaxy input to merge uniprot with genome](../images/2016-11-10_merge-uniprot-with-genome.png)

In [9]:
# The resulting table is data-rich; it has extraneous info (e.g. the contig ID is listed twice), but I will leave as...
# is and reformat later when I want to visualize and make available for others' use

!head -n 2 Geo-70k-scaff-annotations-merged-table.tabular

comp102842_c0_seq1	3.00E-26	O75626	PR domain zinc finger protein 1 (EC 2.1.1.-) (BLIMP-1) (Beta-interferon gene positive regulatory domain I-binding factor) (PR domain-containing protein 1) (Positive regulatory domain I-binding factor 1) (PRDI-BF1) (PRDI-binding factor 1)	Homo sapiens (Human)	aorta development [GO:0035904]; artery morphogenesis [GO:0048844]; cell fate commitment [GO:0045165]; coronary vasculature development [GO:0060976]; eye photoreceptor cell development [GO:0042462]; germ cell development [GO:0007281]; heart valve development [GO:0003170]; intestinal epithelial cell development [GO:0060576]; maternal placenta development [GO:0001893]; morphogenesis of a branching structure [GO:0001763]; negative regulation of B cell proliferation [GO:0030889]; negative regulation of lipopolysaccharide-mediated signaling pathway [GO:0031665]; negative regulation of transcription from RNA polymerase II promoter [GO:0000122]; positive regulation of B cell differentiation [GO:0045579]; 

---

# Identifying transposable elements

To identify transposable elements in my genome (transcriptome?!?!) I will use the RepeatMasker program. RepeatMasker requires Linux operating system and several dependencies. The RepeatMasker download page provided very clear instructions for install and configure.

* [Install RepeatMasker etc.](https://seanb80.github.io/blog/2016/11/04/Installing-RepeatMasker)  
  * Installed [RMBlast](http://www.repeatmasker.org/RMBlast.html) (NCBI Blast modified for use with RepeatMasker/RepeatModeler): 
    * RMBlast Binaries: ftp://ftp.ncbi.nlm.nih.gov/blast/executables/rmblast/2.2.28
    * BLAST+ Binaries: ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/2.2.28/
  * [Installed TRF - Tandem Repeat Finder](http://tandem.bu.edu/trf/trf.html). Version: [Tandem Repeats Finder 4.09 for 64 bit Mac OS X](http://tandem.bu.edu/trf/trf409.macosx.download.html)
  * Installed [RepBase](http://www.girinst.org) for RepeatMasker, database of transposable elements: [repeatmaskerlibraries-20160829.tar.gz (69.69 MB)](http://www.girinst.org/server/RepBase/index.php)
  
For a detailed view of the installation and configure process, see Sean's [Nov. 4 2016 blog post](https://seanb80.github.io/blog/2016/11/04/Installing-RepeatMasker). I found this very helpful to walk me through the process. 

### Running RepeatMasker on Geoduck genome - 70k Scaffolds

RepeatMasker does the following: 

_RepeatMasker is a program that screens DNA sequences for interspersed repeats and low complexity DNA sequences. The output of the program is a detailed annotation of the repeats that are present in the query sequence as well as a modified version of the query sequence in which all the annotated repeats have been masked (default: replaced by Ns)._ -[RepeatMasker website](http://www.repeatmasker.org/webrepeatmaskerhelp.html)

I will follow another [blogpost by Sean](https://seanb80.github.io/blog/2016/11/11/Running-Repeat-Masker-on-Oly-10K-scaffold).

In [2]:
pwd

'/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/Jupyter-Notebooks'

In [3]:
cd ..

/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project


In [4]:
cd analyses

/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/analyses


In [5]:
mkdir RepeatMasker/

In [6]:
cd RepeatMasker

/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/analyses/RepeatMasker


In [1]:
! sudo perl /usr/local/RepeatMasker/RepeatMasker \
-species "crassostrea gigas" -gff -s ../data/Panopea_generosa_scaff-70k.scafSeq

Password:


#### The RepeatMasker program did not want to work via Jupyter, but it DID work very well directly in the terminal. 

I ran RepeatMasker twice. 

**1)**: using the default, "unrefined" speed and **2)** using the slow, "refined" speed which takes ~3 times longer but will find and mask 0-5% more repetitive DNA sequences than the default setting. 

**1)** `! sudo perl /usr/local/RepeatMasker/RepeatMasker \
   -species "crassostrea gigas" -gff -s ../data/Panopea_generosa_scaff-70k.scafSeq`

**2)** `! sudo perl /usr/local/RepeatMasker/RepeatMasker \
   -species "crassostrea gigas" -gff ../data/Panopea_generosa_scaff-70k.scafSeq`
  
Note:
 -species "crassostrea gigas" specifies a species library; if none specified the human genome library is the default
 -gff creates a .gff file in the output
 -sudo overrides permissions/provides access the RepeatMasker program, as it's located in the restricted /usr/local folder
 -RepeatMasker automatically prints to file in the working directory
 
Checking out the results:

In [29]:
!head ../analyses/RepeatMasker/unrefined/Panopea_generosa_scaff-70k.scafSeq.out.gff

##gff-version 2
##date 2016-11-15
##sequence-region Panopea_generosa_scaff-70k.scafSeq
scaffold10970	RepeatMasker	similarity	2053	2317	24.8	+	.	Target "Motif:(CA)n" 1 264
scaffold10970	RepeatMasker	similarity	3020	3051	21.7	+	.	Target "Motif:(TTATAA)n" 1 32
scaffold10970	RepeatMasker	similarity	7560	7949	32.2	-	.	Target "Motif:Tx1-4_CGi" 3372 3696
scaffold10970	RepeatMasker	similarity	10827	10873	 7.1	+	.	Target "Motif:(ATT)n" 1 47
scaffold10970	RepeatMasker	similarity	12491	12558	 0.0	+	.	Target "Motif:(TAAC)n" 1 68
scaffold10970	RepeatMasker	similarity	17918	17949	17.6	+	.	Target "Motif:(TTGTAT)n" 1 33
scaffold10970	RepeatMasker	similarity	19206	19241	19.7	+	.	Target "Motif:(TATA)n" 1 37


In [31]:
!grep -c "^scaffold*" ../analyses/RepeatMasker/unrefined/Panopea_generosa_scaff-70k.scafSeq.out.gff

642


In [30]:
!head ../analyses/RepeatMasker/refined/Panopea_generosa_scaff-70k.scafSeq.out.gff

##gff-version 2
##date 2016-11-15
##sequence-region Panopea_generosa_scaff-70k.scafSeq
scaffold10970	RepeatMasker	similarity	2053	2317	24.8	+	.	Target "Motif:(CA)n" 1 264
scaffold10970	RepeatMasker	similarity	3020	3051	21.7	+	.	Target "Motif:(TTATAA)n" 1 32
scaffold10970	RepeatMasker	similarity	7560	7949	32.2	-	.	Target "Motif:Tx1-4_CGi" 3372 3696
scaffold10970	RepeatMasker	similarity	10195	10460	30.3	-	.	Target "Motif:Tx1-2_CGi" 2686 2956
scaffold10970	RepeatMasker	similarity	10827	10873	 7.1	+	.	Target "Motif:(ATT)n" 1 47
scaffold10970	RepeatMasker	similarity	12491	12558	 0.0	+	.	Target "Motif:(TAAC)n" 1 68
scaffold10970	RepeatMasker	similarity	17918	17949	17.6	+	.	Target "Motif:(TTGTAT)n" 1 33


In [32]:
!grep -c "^scaffold*" ../analyses/RepeatMasker/refined/Panopea_generosa_scaff-70k.scafSeq.out.gff

653


**Not much of a difference between the two results; looks like the "refined" option found 11 more TE sequences**

---

# Identifying candidate miRNA locations

I will use miRBase to identify sequences in genome scaffolds that are likely to code for miRNA.

[miRBase](http://www.mirbase.org/): _The miRBase database is a searchable database of published miRNA sequences and annotation. Each entry in the miRBase Sequence database represents a predicted hairpin portion of a miRNA transcript (termed mir in the database), with information on the location and sequence of the mature miRNA sequence (termed miR). Both hairpin and mature sequences are available for searching and browsing, and entries can also be retrieved by name, keyword, references and annotation. All sequence and annotation data are also available for download._

**Step 1:** download the following [miRBase sequence download](http://www.mirbase.org/ftp.shtml): fasta format sequences of all miRNA hairpins, "hairpin.fa", download date : 11/16/2016

**Step 2:** Let's move the haripin.fa file to the appropriate directory, and then check it out.

In [9]:
cp ../../../../../../Downloads/hairpin.fa.gz ../data/

In [11]:
cd ../data

/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/data


In [14]:
!gunzip hairpin.fa.gz

In [15]:
!head hairpin.fa

>cel-let-7 MI0000001 Caenorhabditis elegans let-7 stem-loop
UACACUGUGGAUCCGGUGAGGUAGUAGGUUGUAUAGUUUGGAAUAUUACCACCGGUGAAC
UAUGCAAUUUUCUACCUUACCGGAGACAGAACUCUUCGA
>cel-lin-4 MI0000002 Caenorhabditis elegans lin-4 stem-loop
AUGCUUCCGGCCUGUUCCCUGAGACCUCAAGUGUGAGUGUACUAUUGAUGCUUCACACCU
GGGCUCUCCGGGUACCAGGACGGUUUGAGCAGAU
>cel-mir-1 MI0000003 Caenorhabditis elegans miR-1 stem-loop
AAAGUGACCGUACCGAGCUGCAUACUUCCUUACAUGCCCAUACUAUAUCAUAAAUGGAUA
UGGAAUGUAAAGAAGUAUGUAGAACGGGGUGGUAGU
>cel-mir-2 MI0000004 Caenorhabditis elegans miR-2 stem-loop


**Step 3:** blastn the hairpin.fa file against my scaffolds

In [18]:
!/Applications/ncbi-blast-2.5.0+/bin/blastn

BLAST query/options error: Either a BLAST database or subject sequence(s) must be specified
Please refer to the BLAST+ user manual.


In [30]:
!/Applications/ncbi-blast-2.5.0+/bin/blastn \
-query ../data/hairpin.fa \
-db ../data/Panopea_generosa_scaff-70k_db \
-task blastn \
-evalue 10e-20 \
-outfmt 6 \
-max_target_seqs 1 \
-num_threads 6 \
-out ../analyses/pgenerosa-scaff70-miRNA.tab

In [34]:
!file ../analyses/pgenerosa-scaff70-miRNA.tab

../analyses/pgenerosa-scaff70-miRNA.tab: empty


In [2]:
#trying again, with a less strict e-value
!/Applications/ncbi-blast-2.5.0+/bin/blastn \
-query ../data/hairpin.fa \
-db ../data/Panopea_generosa_scaff-70k_db \
-task blastn \
-evalue 100e-20 \
-outfmt 6 \
-max_target_seqs 1 \
-num_threads 6 \
-out ../analyses/pgenerosa-scaff70-miRNA.tab

In [3]:
!file ../analyses/pgenerosa-scaff70-miRNA.tab

../analyses/pgenerosa-scaff70-miRNA.tab: empty


In [4]:
#trying again, this time NOT specifying a min e-value 
!/Applications/ncbi-blast-2.5.0+/bin/blastn \
-query ../data/hairpin.fa \
-db ../data/Panopea_generosa_scaff-70k_db \
-task blastn \
-outfmt 6 \
-max_target_seqs 1 \
-num_threads 6 \
-out ../analyses/pgenerosa-scaff70-miRNA.tab

In [5]:
#Viewing file info
!file ../analyses/pgenerosa-scaff70-miRNA.tab

../analyses/pgenerosa-scaff70-miRNA.tab: ASCII text


In [16]:
#viewing file header - the 3rd column, I believe, is the the % match. I'll pare this down later, but first I want to visualize the data.
!head -20 ../analyses/pgenerosa-scaff70-miRNA.tab

cel-let-7	scaffold45727	89.474	19	2	0	57	75	44493	44511	1.5	26.5
cel-lin-4	scaffold4463	100.000	13	0	0	20	32	17268	17280	5.0	24.7
cel-mir-1	scaffold15463	94.737	19	1	0	18	36	47318	47300	0.12	30.1
cel-mir-2	scaffold30278	100.000	16	0	0	2	17	6590	6575	0.12	30.1
cel-mir-34	scaffold27692	100.000	14	0	0	35	48	65424	65437	1.5	26.5
cel-mir-35	scaffold31723	100.000	16	0	0	82	97	26403	26388	0.12	30.1
cel-mir-36	scaffold4463	89.474	19	2	0	74	92	9287	9305	1.5	26.5
cel-mir-37	scaffold26337	100.000	15	0	0	7	21	108320	108334	0.43	28.3
cel-mir-38	scaffold59644	100.000	15	0	0	42	56	21905	21891	0.42	28.3
cel-mir-38	scaffold59644	100.000	13	0	0	42	54	51488	51476	5.1	24.7
cel-mir-39	scaffold18558	100.000	14	0	0	9	22	28960	28947	1.3	26.5
cel-mir-41	scaffold71773	100.000	15	0	0	41	55	87277	87263	0.43	28.3
cel-mir-42	scaffold26337	100.000	14	0	0	77	90	49745	49732	1.5	26.5
cel-mir-43	scaffold32578	84.000	25	4	0	39	63	37653	37677	0.43	28.3
cel-mir-44	scaffold4463	90.000	20	2	0	47	66	46253	46234	0.42	28.3
cel-

In [7]:
#before converting .tab file to .gff file, check out the "2_Blast2Gff.pl" script's manual to confirm commands
! man ../scripts/2_Blast2Gff.pl

/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/Jupyter-Notebooks/../scripts/2_Blast2Gff.pl:176: a tab character is not allowed in an escape name
/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/Jupyter-Notebooks/../scripts/2_Blast2Gff.pl:176: a tab character is not allowed in an escape name
/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/Jupyter-Notebooks/../scripts/2_Blast2Gff.pl:180: cannot use character `*' as a starting delimiter
/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/Jupyter-Notebooks/../scripts/2_Blast2Gff.pl:180: cannot use character `*' as a starting delimiter
#!/usr/bin/perl                                                -w
#-----------------------------------------------------------+   #
| # Blast2Gff.pl                                              | #
| #---------------

In [12]:
#convert to gff format to plot in IGV
!perl ../scripts/2_Blast2Gff.pl \
-i ../analyses/pgenerosa-scaff70-miRNA.tab \
-s "something" \
-o ../analyses/pgenerosa-scaff70-miRNA.gff \
-p "gene-blast" \
-d "mirBase hairpin.fa"

In [9]:
ls ../analyses/

2016-11-3_Geo70k-scaff-transcrip-sequences.tabular
GeoTrans-PhelDiffExp_blasted_sorted
Phel_DEGlist.tab
RepeatMasker/
SeaStarDiffExp.R
pgenerosa-scaff70-miRNA.gff
pgenerosa-scaff70-miRNA.tab
pgenerosa-transcrv3-blastn-scaff70k-01.gff
pgenerosa-transcrv3-blastn-scaff70k-01.gff.fai
pgenerosa-transcrv3-blastn-scaff70k-01.tab
pgenerosa-transcrv3-blastn-scaff70k-01.tab.fai


In [15]:
!head -20 ../analyses/pgenerosa-scaff70-miRNA.gff

scaffold45727	blastn:mirBase hairpin.fa	blastn	44493	44511	1.5	+	.	cel-let-7	
scaffold4463	blastn:mirBase hairpin.fa	blastn	17268	17280	5.0	+	.	cel-lin-4	
scaffold15463	blastn:mirBase hairpin.fa	blastn	47318	47300	0.12	-	.	cel-mir-1	
scaffold30278	blastn:mirBase hairpin.fa	blastn	6590	6575	0.12	-	.	cel-mir-2	
scaffold27692	blastn:mirBase hairpin.fa	blastn	65424	65437	1.5	+	.	cel-mir-34	
scaffold31723	blastn:mirBase hairpin.fa	blastn	26403	26388	0.12	-	.	cel-mir-35	
scaffold4463	blastn:mirBase hairpin.fa	blastn	9287	9305	1.5	+	.	cel-mir-36	
scaffold26337	blastn:mirBase hairpin.fa	blastn	108320	108334	0.43	+	.	cel-mir-37	
scaffold59644	blastn:mirBase hairpin.fa	blastn	21905	21891	0.42	-	.	cel-mir-38	
scaffold59644	blastn:mirBase hairpin.fa	blastn	51488	51476	5.1	-	.	cel-mir-38	
scaffold18558	blastn:mirBase hairpin.fa	blastn	28960	28947	1.3	-	.	cel-mir-39	
scaffold71773	blastn:mirBase hairpin.fa	blastn	87277	87263	0.43	-	.	cel-mir-41	
scaffold26337	blastn:mirBase hairpin.fa	blastn	49745	4

---

# Identifying potential methylation sites, aka CpG sites

Use [Galaxy's](https://usegalaxy.org/) EMBOSS fuzznuc tool to identify CpG sites; these are locations that have potential to be methylated. Select .gff as output file for visualizing in IGV.

![Galaxy-fuzznuc](../images/2016-11-17_Galaxy-fuzznuc.png)

In [17]:
# Resulting file looks like: 
!head ../analyses/Panopea_generosa_Scaff-70k-CpG.gff

##gff-version 2.0
##date 2016-11-17
##Type DNA scaffold3071
scaffold3071	fuzznuc	misc_feature	18	19	2.000	+	.	Sequence "scaffold3071.1" ; note "*pat pattern1"
scaffold3071	fuzznuc	misc_feature	21	22	2.000	+	.	Sequence "scaffold3071.2" ; note "*pat pattern1"
scaffold3071	fuzznuc	misc_feature	23	24	2.000	+	.	Sequence "scaffold3071.3" ; note "*pat pattern1"
scaffold3071	fuzznuc	misc_feature	27	28	2.000	+	.	Sequence "scaffold3071.4" ; note "*pat pattern1"
scaffold3071	fuzznuc	misc_feature	88	89	2.000	+	.	Sequence "scaffold3071.5" ; note "*pat pattern1"
scaffold3071	fuzznuc	misc_feature	188	189	2.000	+	.	Sequence "scaffold3071.6" ; note "*pat pattern1"
scaffold3071	fuzznuc	misc_feature	240	241	2.000	+	.	Sequence "scaffold3071.7" ; note "*pat pattern1"


In [20]:
!grep -c "^scaffold*" ../analyses/Panopea_generosa_Scaff-70k-CpG.gff

42638


---

# Getting a first glimps @ the data via IGV

[Integrative Genomics Viewer (IGV)](http://software.broadinstitute.org/software/igv/), Version 2.3.85 (135)

All scaffolds: 

![IGV 11-17](../images/2016-11-17_IGV.png)  


A closer view at scaffold 71773 (very last one @ far right), which looks interesting:

![IGV 11-17 zoom](../images/2016-11-17_IGV-zoomed.png)